packages/proxy/src/proxy.ts

```diff
@@ -66,13 +68,20 @@ import {
   makeTempCredentials,
   verifyTempCredentials,
 } from "utils";
+import { differenceInSeconds, formatISO, parseISO } from "date-fns";
 import { openAIChatCompletionToChatEvent } from "./providers/openai";
 import { ChatCompletionCreateParamsBase } from "openai/resources/chat/completions";
 import { importPKCS8, SignJWT } from "jose";
 import { z } from "zod";
 
+type CachedMetadata = {
+  cached_at: string;
+  ttl: number;
```

Comment:
reflected by another comment but we could alternatively parameterize this as `(cachedAt, expiration)` which may be easier to work with